src/Utilities/CssConcatenator/RuleBlock.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Utilities\CssConcatenator;
+
+/**
+ * This class represents a CSS rule block.
+ *
+ * @internal
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ * @author SignpostMarv
+ * @author Oliver Klee <github@oliverklee.de>
+ */
+class RuleBlock
+{
+    /**
+     * array whose keys are selectors for the rule block (values are of no significance)
+     *
+     * @var array
+     *
+     * @psalm-var array<string, mixed>
+     */
+    private $selectorsAsKeys = [];
+
+    /**
+     * the property declarations, e.g. "margin-top: 0.5em; padding: 0"
+     *
+     * @var string
+     */
+    private $declarationsBlock;
+
+    /**
+     * @param array $selectorsAsKeys
+     * @param string $declarationsBlock
+     *
+     * @psalm-param array<string, mixed> $selectorsAsKeys
+     */
+    public function __construct(array $selectorsAsKeys, string $declarationsBlock)
+    {
+        $this->selectorsAsKeys = $selectorsAsKeys;
+        $this->declarationsBlock = $declarationsBlock;
+    }
+
+    /**
+     * @return array
+     *
+     * @psalm-return array<string, mixed>
+     */
+    public function getSelectorsAsKeys(): array
+    {
+        return $this->selectorsAsKeys;
+    }
```

Comment:
With the suggested refactoring, we'll probably prefer instead a method `getSelectors` which returns `\array_keys($this->selectorsAsKeys)`.